File: puml_server/storage/4179d4b1b44241e2905b5f38a8ea0081/plugin_outputs/3c027ed02dbf4e40a2d50e3167f3d71a/classes.cpp
#include <string>
#include <vector>

class Book;
class Library;
class Reader;


class Book {
public:
    void borrow();
    void returnBook();
private:String isbn;
    String title;
    String author;
    boolean isAvailable;
    };

class Library {
public:
    void addBook(Book book);
    Book findBook(String title);
    List<Book> listAvailableBooks();
private:String name;
    String address;
    };

class Reader {
public:
    boolean borrowBook(Book book);
    void returnBook(Book book);
private:int readerId;
    String name;
    String email;
    };

File: puml_server/storage/4179d4b1b44241e2905b5f38a8ea0081/plugin_outputs/3c027ed02dbf4e40a2d50e3167f3d71a/classes.py
from typing import List


class Book:
    def __init__(self, isbn: String, title: String, author: String, isAvailable: boolean):
        
        
        self.isbn: String = isbn
        
        self.title: String = title
        
        self.author: String = author
        
        self.isAvailable: boolean = isAvailable
        

    
    def borrow(self) -> 'void':
        pass
    
    def returnBook(self) -> 'void':
        pass
    

class Library:
    def __init__(self, name: String, address: String):
        
        
        self.name: String = name
        
        self.address: String = address
        

    
    def addBook(self, book: Book) -> 'void':
        pass
    
    def findBook(self, title: String) -> 'Book':
        pass
    
    def listAvailableBooks(self) -> 'List<Book>':
        pass
    

class Reader:
    def __init__(self, readerId: int, name: String, email: String):
        
        
        self.readerId: int = readerId
        
        self.name: String = name
        
        self.email: String = email
        

    
    def borrowBook(self, book: Book) -> 'boolean':
        pass
    
    def returnBook(self, book: Book) -> 'void':
        pass
    
